### ISSUE TITLE
ALEX - research on innovative lending protocol that eliminates liquidation risk

### ISSUE THREAD

**Issue by fiftyeightandeight** [stacksgov/Stacks-Grants#194]
## Background
Protocols for loanable funds (PLF) enable borrowing and lending activities. Examples PLFs are Compound and Aave on Ethereum. The lender provides a token in need and earns interest in return. The borrower deposits collateral and gets access to a preferred asset. The borrower must pay back the borrowed asset in due time. Protocols enabling this borrowing and lending functionality are incredibly useful. Simply, they enable the present consumption on future earnings. This idea is powerful, and has been at the core of DeFi's rise, including the rise of protocols such as Uniswap.

However, one of the risks posed to market participants in existing PLFs is default risk. While each loan must be secured with collateral, the price of crypto collateral can fluctuate wildly and quickly. As a result, many PLFs ask borrows to significantly overcollateralise their positions. Overcollateralisation refers to value of collateralised assets being higher than the value of the loaned assets. The proportion of the collateral value to loaned value is often called "collateralisation ratio" (CR), the inverse of "loan to value" (LTV). For simplicity sake, we use the term LTV throughout the paper. The higher the LTV is, the more likely the default occurs. A LTV larger than 1 means the value of collateral cannot cover the value of the loaned asset.

## Research Overview
In variable rate platforms, such as Aave, collateral in the form of more liquid assets tends to have a higher LTV. In fixed-rate fixed-term protocols, such as YieldSpace, using ETH as collateral to borrow Dai requires a LTV of 67%. In the event that the portfolio is underfunded, three scenarios could emerge in the existing protocols: (i) a borrower could top up the collateral asset to stay afloat; (ii) a borrower could return some of the borrowing asset to decrease the LTV; and (iii) the loan could be unwound by a third party such as liquidator if the borrower defaults. A third party unwinds a borrower's position by paying back the loan, and in return earns certain fees. In cases when a collateral asset is illiquid, fees can be as high as 15% on Aave, representing a significant penalty to defaulting borrowers. This also poses disruption to borrowing/lending activity, as a pre-agreed loan is terminated early.

Our research aims to abolish liquidation.We aim to keep the loan active until maturity, regardless of market condition, solving problems plaguing many existing PLFs. We believe a solution can be found on an innovative combination of asset management and collateral pools.

## Deliverables
- Deliverable: a whitepaper on Collateral Rebalancing Pool. The deliverable will be available as an open source paper.
- Total requested grant for this deliverable = $10,000
- Deliverable deadline: Oct 30

## Community and Supporting Materials
Two projects of Stacks Accelerator Cohort 1 indicated so far that they would be keen to integrate the protocol developed based on the research into their own product offering. We would be happy to share more details with the Foundation.

**Comment by jennymith** [comment]
Hi @fiftyeightandeight, thanks for your submission! This is an interesting research area but one that explores a well-known, well-researched problem with no viable solution yet. 

**Decision:** While your solution sounds promising, it's not clear from your proposal _how_ this research would advance or improve upon the current discourse around liquidation risk (tagging @jcnelson to share relevant/existing work). **We have decided to pass on this proposal for now but would encourage you to continue refining these ideas!**

Here are some of the questions/points of feedback that came up in our discussion:
- How does the ALEX team actually plan to conduct research? What is their methodology?
- They mention that they may be able to abolish liquidation through an innovative combination of asset management and collateral pools–can they elaborate on this?
- In other words, what is the hypothesis that they plan to test?
- How is this work different from the research that's already out there? (What have they discovered so far that others have not?)
- Given that DeFi on Stacks is still very early, would this research be immediately valuable to community members? In what ways would it be valuable?
- Are there any builders or teams who would be willing to vouch for the demand for this research and explain how it would help them in their product offerings?

**Comment by fiftyeightandeight** [comment]
Hi @jennymith thank you for the feedbacks and we respect the committee's decision. We will consider your questions/feedbacks and come back if we have any follow-ups.

**Comment by tycho1212** [comment]
@fiftyeightandeight I've been studying your whitepaper in quite some depth recently and I just felt compelled to give my thoughts on this :)

It seems to me like this question around liquidation-free loans can best be solved through practice rather than research. I think your whitepaper is very comprehensive around this area already. The main question mark seems to be how the ALEX reserve fund will be capitalized during a drawdown. You will need pretty much infinite funds to save the protocol in that case. There are some proven practices from the DeFi community that solve for problems like this.

MakerDAO solves a similar problem by printing & auctioning additional governance tokens in case the protocol goes underwater. In theory this gives them infinite liquidity in case of a drawdown, unless people lose trust in Maker and the price of the goverance token goes to 0. Would be happy to share more in case you're interested!

**Comment by fiftyeightandeight** [comment]
Hi @tycho1212, thank you for your comments. I assume you already took a look at our [whitepaper](https://docs.alexgo.io/whitepaper/automated-market-making-of-collateral-rebalancing-pool) on this topic.

Yes, you are absolutely right. How the ALEX reserve fund is capitalised during a drawdown is of paramount importance to the confidence of our community in ALEX.

And it is not just about having a reserve fund, but much more, including (not exhaustive) how it is created, what composes the reserve fund, how it is managed, what ratio is appropriate for the size of the reserve fund vs. the outstanding obligation (i.e. loan), etc.

The management of the reserve fund must be guided and driven by its primary objective (of acting as the last resort) and its size must then dictates the size of the risks outstanding on ALEX.

Our team, as a follow-up to the whitepaper I mentioned above, is looking into this and will publish relevant reports in due course. We would be very happy to discuss some early details with you, if you are interested!

On printing & auctioning additional governance token, we saw protocols like AAVE also utilise this concept, but this particular mechanism is not something we like. While theoretically possible, this is a wrong-way risk (you print precisely when you are weak and can't afford) that can be very dangerous to a platform should such an option be exercised. So our approach is to make sure the reserve fund is created/managed in a robust way so we do not need infinite liquidity and do not need to introduce such wrong-way risk into the system.

Thank you.

**Comment by tycho1212** [comment]
> We would be very happy to discuss some early details with you, if you are interested!
I'd be very curious! It's still hard for me to see how you could manage a reserve fund in such a way that you can eliminate the option for infinite liquidity. I'll send you a dm :)
